### DISCUSSION TITLE
Horizontal scroll for preview?

### DISCUSSION THREAD

**Discussion by simonhf** [alagrede/znote-app#50] [General]
If I created a mark down block wider than the preview, e.g.:

```
this ... might ... be ...                       ... very wide ...        and wider ...      ...than the preview width ...
```

How to enable a horizontal scroll bar to view the non-wrapping text off the right edge?

**Comment by alagrede** [comment]
Hello @simonhf,
Thank you for your feedback. Currently, the behavior is: 
- On Windows: the horizontal scrollbar is shown
- With Mac the horizontal scrollbar is hidden (but you can still slide with the touchpad)
The correct behavior on Mac should be to automatically display the horizontal scrollbar when the cursor hovers over the code block
If that's the problem, I'll fix it in my next release. 👍

Thank you
Anthony

**Reply by simonhf** [reply]
Thanks for this info.

Yes, I'm running on Mac and the scrollbar must be always hidden :-(

Unfortunately I mostly use the Mac with external peripherals so the touchpad option isn't so useful for me :-(

Thanks for addressing this issue in the next release which I'll keep an eye out for! :-)

**Comment by alagrede** [comment]
Horizontal scroll bar fix is released in version 1.3.0 😊
I hope this will help you

Anthony

**Reply by simonhf** [reply]
Thanks for the update @alagrede! When will version 1.3.0 be released? Releases [1] only shows version 1.2.0 and the Mac app store only shows version 1.1.9...

[1] https://github.com/alagrede/znote-app/releases

**Comment by alagrede** [comment]
Yes sorry, I always update first Znote Lab here: https://znote.io/
The regular version of Znote and App stores have a slower release lifecycle because I need to wait each store validation to synchronize versions. It generally take 1-2 more weeks.
I hope you understand.

**Comment by simonhf** [comment]
@alagrede Thanks for the extra info! :-) I'll give it a try :-)

**Comment by simonhf** [comment]
@alagrede I really like the new output panel for navigating the outline, but I have two issues with it:
1. When it's opened then it overlaps with the rendered markdown (right most) pane.
2. It's not possible to vary the width of the outline panel, and I have a lot wider header titles which wrap and cause the output to look not very good :-( Example:

![image](https://user-images.githubusercontent.com/357158/151042916-19ba8969-e85b-4894-befd-a983cfddacef.png)

Here's my wish list for tweaking the outline :-)

- When the outline is open then push over the panels to the left so there is no overlapping.
- Have the width of the outline panel changeable so that the user has control over how much or little horizontal header title real estate is visible.
- Have the header titles so that they don't wrap and take up valuable vertical real estate causing more vertical scrolling to navigate.

Also, is it possible to change the behaviour of the main editing window to enable or disable wrapping of longer lines somehow? As you can see in my screen shot, the lines are being wrapping and it makes the lines more confusing and difficult to navigate and edit.

Hope this helps! And thanks again for adding this feature! :-)

**Reply by alagrede** [reply]
Haha it seems I still have lot of work to do 😝. I'm happy you find the new outline panel useful. Thanks to your feedback, I figure out your issue. I will update the panel to be more readable with long titles in my next releases. 
Thank you @simonhf


Anthony

**Reply by simonhf** [reply]
Thanks @alagrede  ! I'll keep an eye out for new releases :-)

**Reply by simonhf** [reply]
@alagrede... I was just checking [1] for new releases and happy to see so many new releases :-) I hope the features above will make it in at some point because they are the only thing stopping me from switching to znote-app :-)

[1] https://github.com/alagrede/znote-app/releases

**Comment by alagrede** [comment]
Hi @simonhf, thanks for your feedback. We are continuously improving the app 😊
To summarize the done and remaining work here: 
- The horizontal scrollbar is fixed on Mac
- Outline panel improved to avoid horizontal scroll bar and add text ellipsis on long title
With the current state, the outline panel makes it easy to navigate through the note.

Resizing the outline panel and giving it a static position next to the note content without overlapping is a lot trickier to do. But it will come!

Thank you for your understanding.
Anthony

**Reply by simonhf** [reply]
The outline panel looks so much neater now... thanks! :-)

![image](https://user-images.githubusercontent.com/357158/159764458-cf5cb11e-08d5-4e04-8de9-69db65d83d26.png)

**Reply by alagrede** [reply]
Thank you 😊

**Comment by alagrede** [comment]
The related issue: https://github.com/alagrede/znote-app/issues/57

**Comment by simonhf** [comment]
> Also, is it possible to change the behaviour of the main editing window to enable or disable wrapping of longer lines somehow? As you can see in my screen shot, the lines are being wrapped and it makes the lines more confusing and difficult to navigate and edit.

BTW this question fell between the cracks somehow :-) Ideally there would be an easy way somehow to toggle between line wrapping and non-wrapping? I'm also not sure if a non-wrapping editing mode exists for the main editing window :-)

**Reply by alagrede** [reply]
https://github.com/alagrede/znote-app/issues/58
